src/turbomind/api/python/linear.cc

```diff
@@ -70,6 +76,12 @@ struct Linear::Impl {
         convert_scales_zeros(workspace, scales, qzeros, input_dims_, output_dims_, group_size_, simt);
 
         check_cuda_error(cudaFree(workspace));
+
+        // try to create a gemm workspace for <device_id, default_stream>
+        //and cache it to `workspace_cache_`
+        int device_id;
+        check_cuda_error(cudaGetDevice(&device_id));
+        (void)getWorkspace(device_id, 0);
```

Comment:
Why do we need this?

Reply:
If the default CUDA stream is used, I guess the forward pass could benefit from pre-creating the workspace.